### ISSUE TITLE
PowerSet function is setup to use arrays instead of lists

### ISSUE THREAD

**Issue by stevk** [bghgary/glTF-Asset-Generator#9]
- [x] Convert the function to use lists
- [x] Make code changes as required, so that parameters are stored in lists everywhere else
- [x] Is there a perf gain by switching to lists? (likely yes, due to resizing)

**Comment by stevk** [comment]
Try this code from kcoley 

  ```
      public static T[][] PowerSet<T>(T[] seq)
        {
            // Create and initialize List of List<T>s
            var powerSet = new List<List<T>>((1 << seq.Length));
            for (int i = 0; i < (1 << seq.Length); ++i)
            {
                powerSet.Add(new List<T>());
            }

            for (int i = 0; i < seq.Length; ++i)
            {
                var cur = seq[i];
                int count = 1 << i;
                for ( int j = 0; j < count; ++j)
                {
                    var source = powerSet[j];

                    // Create and initialize List of size source.Count + 1
                    var destination = powerSet[count + j] = new List<T>(source.Count + 1);
                    for (int q = 0; q < (source.Count + 1); ++q)
                        destination.Add(default(T));
                    
                    for (int q = 0; q < (source.Count); ++q)
                        destination[q] = source[q];
                   
                    destination[source.Count] = cur;
                }
            }
            return powerSet.Select(a => a.ToArray()).ToArray();
        }
```

**Comment by stevk** [comment]
This is the solution (although there might be some fat here that could be trimmed). There is a perf gain, but very slight. The building of the PowerSet takes slightly longer with lists, but we save time overall by not converting from array to list later (which is needed in order to trim unwanted combos). 

        public static List<List<T>> PowerSet<T>(List<T> seq)
        {
            var powerSet = new List<List<T>>();
            powerSet.Add(new List<T>()); // starting only with empty set
            for (int i = 0; i < seq.Count; i++)
            {
                var cur = seq[i];
                int count = 1 << i; // doubling list each time
                for (int j = 0; j < count; j++)
                {
                    var source = powerSet[j];
                    powerSet.Add(new List<T>());
                    powerSet[count + j] = new List<T>();
                    var destination = powerSet[count + j];
                    for (int q = 0; q < source.Count; q++)
                        destination.Add(source[q]);
                    destination.Add(cur);
                }
            }

            return powerSet;
        }

**Comment by stevk** [comment]
Fixed by PR #48
